src/xenia/kernel/xam/xam_ui.cc

```diff
@@ -819,38 +821,286 @@ dword_result_t XamShowMarketplaceDownloadItemsUI_entry(
+        }
+
+        ImGui::EndGroup();
+        if (i != (users_needed_ - 1) && (i == 0 || i == 2)) {
+          ImGui::SameLine();
+        }
+      }
+
+      ImGui::Spacing();
+
+      if (ImGui::Button("Create Profile")) {
+        creating_profile_ = true;
+        ImGui::OpenPopup("Create Profile");
+        first_draw = true;
+      }
+      ImGui::Spacing();
+
+      if (creating_profile_) {
+        if (ImGui::BeginPopupModal("Create Profile", nullptr,
+                                   ImGuiWindowFlags_AlwaysAutoResize)) {
+          if (first_draw) {
+            ImGui::SetKeyboardFocusHere();
+          }
+
+          ImGui::TextUnformatted("Gamertag:");
+          ImGui::InputText("##Gamertag", gamertag_, sizeof(gamertag_));
+
+          const std::string gamertag_string = gamertag_;
+          bool valid = profile_manager->IsGamertagValid(gamertag_string);
+
+          ImGui::BeginDisabled(!valid);
+          if (ImGui::Button("Create")) {
+            profile_manager->CreateProfile(gamertag_string, false);
+            std::fill(std::begin(gamertag_), std::end(gamertag_), '\0');
+            ImGui::CloseCurrentPopup();
+            creating_profile_ = false;
+            ReloadProfiles(false);
+          }
+          ImGui::EndDisabled();
+          ImGui::SameLine();
+
+          if (ImGui::Button("Cancel")) {
+            std::fill(std::begin(gamertag_), std::end(gamertag_), '\0');
+            ImGui::CloseCurrentPopup();
+            creating_profile_ = false;
+          }
+
+          ImGui::EndPopup();
+        } else {
+          creating_profile_ = false;
+        }
+      }
+
+      if (ImGui::Button("OK")) {
+        std::map<uint8_t, uint64_t> profile_map;
+        for (uint32_t i = 0; i < users_needed_; i++) {
+          uint8_t slot = chosen_slots_[i];
+          uint64_t xuid = chosen_xuids_[i];
+          if (slot != 0xFF && xuid != 0) {
+            profile_map[slot] = xuid;
+          }
+        }
+        profile_manager->LoginMultiple(profile_map);
+
+        ImGui::CloseCurrentPopup();
+        Close();
+      }
+      ImGui::SameLine();
+
+      if (ImGui::Button("Cancel")) {
+        ImGui::CloseCurrentPopup();
+        Close();
+      }
+
+      ImGui::Spacing();
+      ImGui::Spacing();
+      ImGui::EndPopup();
+    } else {
+      Close();
+    }
+  }
+
+  void ReloadProfiles(bool first_draw) {
+    auto profile_manager = kernel_state()->xam_state()->profile_manager();
+    auto profiles = profile_manager->GetProfiles();
+
+    profile_data_.clear();
+    for (auto& [xuid, account] : *profiles) {
+      profile_data_.push_back({xuid, account.GetGamertagString()});
+    }
+
+    if (first_draw) {
+      // If only 1 user is requested, request last used controller to sign in.
+      if (users_needed_ == 1) {
+        chosen_slots_[0] = last_user_;
+      } else {
+        for (uint32_t i = 0; i < users_needed_; i++) {
+          chosen_slots_[i] = i;
```

Comment:
Not sure about this, needs to be tested on real hardware but that requires at least three controllers.